main.js

```diff
@@ -297,3 +310,15 @@ function addFileReadError(text, error) {
     text: `${text}<br /><code>${error.toString()}</code>`
   });
 }
+
+/**
+ * Generates the options to be given to render modules.
+ * @param {!Request} request The HTTP request object transmitted by express.
+ */
+function getOptions(request) {
```

Comment:
If you have this function, then you can also incorporate `app.engine('js', ...)`'s `opts` object here instead of having it twice. Or you incorporate this function there and always just pass the `request` object?

Reply:
The options in `app.engine('js', ...)` are static, the options `getOptions(request)` depend on the request and therefore change with each request.

Reply:
Yes, as I said, it would be an idea to also pass the request object to `app.engine(...)` so that the "dynamic" options can also have their default values there. Why should we separate dynamic and static ones?

I just don't see the value in having to use `Object.assign` twice if we almost never overwrite default values, but only add additional ones.

Reply:
Done.